vector/src/main/java/im/vector/app/features/home/room/list/home/HomeLayoutPreferences.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.list.home
+
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import im.vector.app.core.di.DefaultPreferences
+import javax.inject.Inject
+
+class HomeLayoutPreferences @Inject constructor(
```

Comment:
I think it would be easier to use a `DataStore` rather than `SharedPreferences`. See `VectorSessionStore` or `AnalyticsStore` for some example.